game/kernel/asm_funcs.asm

```diff
@@ -436,6 +436,90 @@ _call_goal_on_stack_asm_linux:
   pop r13
   ret
 
+;; The _call_goal_asm function is used to call a GOAL function from C.
+;; It calls on the parent stack, which is a bad idea if your stack is not already a GOAL stack.
+;; It supports up to 3 arguments and a return value.
+;; This should be called with the arguments:
+;; - first goal arg
+;; - second goal arg
+;; - third goal arg
+;; - address of function to call
+;; - address of the symbol table
+;; - GOAL memory space offset
+
+global _call_goal_asm_macos_x64
+
+;; FIXME - copied from linux, but not really tested
+_call_goal_asm_macos_x64:
```

Comment:
As I understand it both MacOS and Linux use the System V calling convention, maybe rename the Linux variants and share it on both platforms?

Reply:
Aha, yeah it seems so:
https://en.wikipedia.org/wiki/X86_calling_conventions#System_V_AMD64_ABI
(Programming at this low level is very much not my forte, so I've been guessing a lot for the ASM integration parts.)

I'll go ahead and suggest a rename in this PR then.